lib/setopt.c

```diff
@@ -3170,6 +3170,16 @@ CURLcode Curl_vsetopt(struct Curl_easy *data, CURLoption option, va_list param)
   case CURLOPT_QUICK_EXIT:
     data->set.quick_exit = (0 != va_arg(param, long)) ? 1L:0L;
     break;
+  case CURLOPT_BLOCK_DOMAIN:
+    /*
+     * A string that defines a domain to block
+     */
+    argptr = va_arg(param, char *);
```

Comment:
Generally libcurl doesn't require the caller to keep the passed parameters such as strings in memory after `curl_easy_setopt`. Thus this is not correct in a way that it doesn't clone the passed string.

```Strings passed to libcurl as 'char *' arguments, are copied by the library; the string storage associated to the pointer argument may be discarded or reused after curl_easy_setopt returns.```
source: https://curl.se/libcurl/c/curl_easy_setopt.html